From f9b7292e637092d36c68123f081729be3a0070ba Mon Sep 17 00:00:00 2001
From: "Tekriwal, Tanuj" <tanuj.tekriwal@intel.com>
Date: Tue, 4 Feb 2025 11:57:05 +0000
Subject: [PATCH] mem_fun_ref is deprecated

mem_fun_ref replace with mem_fn as former is deprecated

Tests Done: Build proceeds

Tracked-On: NA
Signed-off-by: Tekriwal, Tanuj <tanuj.tekriwal@intel.com>
---
 .../mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp  | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp b/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp
index cc3f3e49..f9b7f9c2 100644
--- a/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp
+++ b/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp
@@ -28,7 +28,6 @@
 #include <math.h>
 #include <limits.h> /* for INT_MIN, INT_MAX, etc. on Linux/Android */
 
-
 #include "cmrt_cross_platform.h"
 
 #include <assert.h>
@@ -1907,7 +1906,7 @@ mfxStatus TaskManager::AssignTask(
         std::vector<Surface>::iterator it = std::find_if(
             m_bitstreams.begin(),
             m_bitstreams.end(),
-            std::mem_fun_ref(&Surface::IsFree));
+            std::mem_fn(&Surface::IsFree));
         if (it == m_bitstreams.end())
             return MFX_WRN_DEVICE_BUSY;
 
@@ -1918,7 +1917,7 @@ mfxStatus TaskManager::AssignTask(
             std::vector<Surface>::iterator it2 = std::find_if(
                 it + 1,
                 m_bitstreams.end(),
-                std::mem_fun_ref(&Surface::IsFree));
+                std::mem_fn(&Surface::IsFree));
             if (it2 == m_bitstreams.end())
                 return MFX_WRN_DEVICE_BUSY;
 
-- 
2.34.1

